### PR TITLE
Update SDK layout

### PR DESCRIPTION
**Pull request by lpusok**
### Context

Use ANDROID_SDK_ROOT variable to locate the SDK.
Look up command line tools under `cmdlin-tools/latest/bin`.

Resolves https://bitrise.atlassian.net/browse/STEP-247
### Changes

- added sdk.NewDefaultModel to locate the SDK (by reading the values ANDROID_SDK_ROOT in addition of ANDROID_HOME)
- Look up sdkmanager and avdmanager under `cmdline-tools/latest/bin` in addition to `tools/bin`.

### CODE REVIEW

File: sdk/sdk.go [hunk 1]
```diff
@@ -87,3 +139,39 @@ func (model *Model) LatestBuildToolPath(name string) (string, error) {
...
+			return "", fmt.Errorf("failed to locate Android command-line tools directory, invalid patterns specified (%s): %s", toolPaths, err)
+		}
+
+		if len(matches) == 0 {
+			continue
+		}
+
+		sdkmanagerPath := matches[0]
+
+		fileInfo, err := os.Stat(sdkmanagerPath)
```
Comment (godrei) [inline sdk/sdk.go:164]: previously `pathutil.IsDirExists()` was used for checking if a dir exists

**Review by godrei — CHANGES_REQUESTED** [review]

File: sdk/sdk_test.go [hunk 2]
```diff
@@ -74,3 +74,146 @@ func TestLatestBuildToolPath(t *testing.T) {
...
+			want: &Model{
+				androidHome: sdkRoot,
+			},
+		},
+		{
+			name:    "neither ANDROID_HOME, ANDROID_SDK_ROOT set",
+			envs:    Environment{},
+			want:    nil,
+			wantErr: true,
+		},
```
Comment (godrei) [inline sdk/sdk_test.go:146]: might worth to test an invalid sdk path too

File: sdk/sdk_test.go [hunk 3]
```diff
@@ -74,3 +74,146 @@ func TestLatestBuildToolPath(t *testing.T) {
...
+			},
+			wantPath: "cmdline-tools/latest/bin",
+		},
+		{
+			name: "Command-line tools fixed version",
+			SDKlayout: []string{
+				"cmdline-tools/1.0/bin",
+			},
+			wantPath: "cmdline-tools/1.0/bin",
+		},
```
Comment (godrei) [inline sdk/sdk_test.go:190]: suggest adding a test case for error handling too

File: sdkmanager/sdkmanager.go [hunk 4]
```diff
@@ -17,34 +17,35 @@ type Model struct {
 	binPth      string
 }
 
-// IsLegacySDKManager ...
-func IsLegacySDKManager(androidHome string) (bool, error) {
```
Comment (godrei) [inline sdkmanager/sdkmanager.go:21]: Is not it used by other codes?

File: sdk/sdk.go [hunk 5]
```diff
@@ -87,3 +139,39 @@ func (model *Model) LatestBuildToolPath(name string) (string, error) {
...
+		if err != nil {
+			return "", fmt.Errorf("failed to get file info for (%s): %v", sdkmanagerPath, err)
```
Comment (mateherber) [inline sdk/sdk.go:166]: Shouldn't we just continue and keep trying with others?

File: sdk/sdk_test.go [hunk 6]
```diff
@@ -74,3 +74,146 @@ func TestLatestBuildToolPath(t *testing.T) {
...
+
+func TestModel_CmdlineToolsPath(t *testing.T) {
+	tests := []struct {
+		name      string
+		SDKlayout []string
+		wantPath  string
+		wantErr   bool
+	}{
+		{
+			name: "Tools",
```
Comment (mateherber) [inline sdk/sdk_test.go:170]: legacy just "tools" is not covered?

**Review by mateherber — APPROVED** [review]

File: sdkmanager/sdkmanager.go (hunk 4 above)
Comment (lpusok) [reply sdkmanager/sdkmanager.go:21]: no

**Review by mateherber — APPROVED** [review]

**Pull request merged**